mesonbuild/build.py

```diff
@@ -1160,12 +1161,21 @@ def __init__(self, name, subdir, subproject, is_cross, sources, objects, environ
         # The import library that GCC would generate (and prefer)
         self.gcc_import_filename = None
 
+        # check for modularity
+        self.is_modular = False
+        if 'modular' in kwargs and kwargs['modular']:
+            self.is_modular = True
+        if 'implib' in kwargs and kwargs['implib']:
+            self.is_modular = True
+        if self.is_modular and ('implib' in kwargs and kwargs['implib'] == False):
+            raise InvalidArguments('"implib" must not be set to false for modular executables')
+
         # if implib appears, this target is linkwith:-able, but that only means
         # something on Windows platforms.
         self.is_linkwithable = False
-        if 'implib' in kwargs and kwargs['implib']:
+        if self.is_modular:
             implib_basename = self.name + '.exe'
-            if not isinstance(kwargs['implib'], bool):
+            if kwargs.get('implib') != None and not isinstance(kwargs['implib'], bool):
```

Comment:
__[Flake8]__

comparison to None should be 'if cond is not None:'


<!-- Comment by [SideCI](https://sideci.com) -->